rootston/cursor.c

```diff
@@ -76,27 +78,153 @@ static void cursor_set_xcursor_image(struct roots_input *input,
 	}
 }
 
+static struct wl_list *surface_layers_update_position(
+		struct roots_input *input, struct wl_list *layer_surfaces, double x,
+		double y, uint32_t time, enum surface_layers_layer until_layer,
+		struct wl_list **focused_list, struct wlr_layer_surface **exclusive,
+		double *exclusive_sx, double *exclusive_sy) {
+	struct wlr_layer_surface *layer_surface;
+	struct roots_focused_layer_surface *focused_layer_surface;
+	wl_list_for_each_reverse(layer_surface, layer_surfaces, link) {
+		if (layer_surface->layer < until_layer) {
+			return layer_surface->link.next;
+		}
+		if (!(layer_surface->current->input_types &
+				LAYER_SURFACE_INPUT_DEVICE_POINTER)) {
+			continue;
+		}
+
+		if (*focused_list != &input->cursor_focused_layer_surfaces) {
+			focused_layer_surface = wl_container_of(*focused_list,
+				focused_layer_surface, link);
+		} else {
+			// Head of list
+			focused_layer_surface = NULL;
+		}
+
+		struct wl_client *client =
+			wl_resource_get_client(layer_surface->resource);
+		struct wlr_seat_handle *handle =
+			wlr_seat_handle_for_client(input->wl_seat, client);
+
+		double sx, sy;
+		bool ok = layer_surface_is_at(input->server->desktop, layer_surface,
```

Comment:
Variable names like "ok" should be avoided. They always trip me up when I'm reading code.

Reply:
That's a bit pedantic.